internal/acceptance/lakeview_dashboard_test.go

```diff
@@ -0,0 +1,481 @@
+
+	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/service/dashboards"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/databricks/terraform-provider-databricks/qa"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type TemplateStruct struct {
+	DisplayName         string
+	WarehouseId         string
+	ParentPath          string
+	SerializedDashboard string
+	FilePath            string
+	EmbedCredentials    string
+}
+
+func MakeTemplate(template TemplateStruct) string {
+	templateString := fmt.Sprintf(`
+	resource "databricks_lakeview_dashboard" "d1" {
+		display_name			= 	"%s"
+		warehouse_id			=	"%s"
+		parent_path				= 	"%s"
+	`, template.DisplayName, template.WarehouseId, template.ParentPath)
+	if template.SerializedDashboard != "" {
+		templateString += fmt.Sprintf(`	serialized_dashboard	=	"%s"
+	`, template.SerializedDashboard)
+	}
+	if template.FilePath != "" {
+		templateString += fmt.Sprintf(`	file_path				=	"%s"
+	`, template.FilePath)
+	}
+	if template.EmbedCredentials != "" {
+		templateString += fmt.Sprintf(`	embed_credentials		=	"%s"
+	`, template.EmbedCredentials)
+	}
+	templateString += `}
+	`
+	return templateString
+}
+
+// Altough EmbedCredentials is an optional field, please specify its value if you want to modify it.
+func (t *TemplateStruct) SetAttributes(mapper map[string]string) TemplateStruct {
+	// Switch case for each attribute. If it is set in the mapper, set it in the struct
+	if val, ok := mapper["display_name"]; ok {
+		t.DisplayName = val
+	}
+	if val, ok := mapper["warehouse_id"]; ok {
+		t.WarehouseId = val
+	}
+	if val, ok := mapper["parent_path"]; ok {
+		t.ParentPath = val
+	}
+	if val, ok := mapper["serialized_dashboard"]; ok {
+		t.SerializedDashboard = val
+	}
+	if val, ok := mapper["file_path"]; ok {
+		t.FilePath = val
+	}
+	if val, ok := mapper["embed_credentials"]; ok {
+		if val == "true" {
+			t.EmbedCredentials = ""
+		} else {
+			t.EmbedCredentials = val
+		}
+	}
+	return *t
+}
+
+func TestAccLakeviewDashboard(t *testing.T) {
+	var template TemplateStruct
+	workspaceLevel(t, step{
+		Template: MakeTemplate(template.SetAttributes(map[string]string{
+			"display_name":         "Monthly Traffic Report",
+			"warehouse_id":         "{env.TEST_DEFAULT_WAREHOUSE_ID}",
+			"parent_path":          "/Shared/provider-test",
+			"serialized_dashboard": `{\"pages\":[{\"name\":\"new_name\",\"displayName\":\"New Page\"}]}`,
+		})),
+		Check: resourceCheck("databricks_lakeview_dashboard.d1", func(ctx context.Context, client *common.DatabricksClient, id string) error {
+			w, err := client.WorkspaceClient()
+			if err != nil {
+				return err
+			}
+			dashboard, err := w.Lakeview.Get(ctx, dashboards.GetDashboardRequest{
+				DashboardId: id,
+			})
+			if err != nil {
+				return err
+			}
+			assert.Equal(t, "Monthly Traffic Report", dashboard.DisplayName)
+			fmt.Println(dashboard.SerializedDashboard)
```

Comment:
Let's not print the dashboard. In case of test failure, the http calls are printed, so we will see the dashboard.

Reply:
So what can we check in the 'check' function that would be relevant?

Reply:
Not sure. First I would think about validating the Dashabord. But service team does not promise a static format. Maybe check that the length of the serialized dashabord is > 10 (to ensure it's not empty). And add a note